.github/DEVELOPMENT.md

```diff
@@ -210,5 +210,37 @@ The `pnpm run test:migrate` script is run in CI to ensure that templating change
 See `.github/workflows/test-migrate.yml`.
 
 > Tip: if the migration test is failing in CI and you don't see any errors, try [downloading the full logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/using-workflow-run-logs#downloading-logs).
-> There'll likely be a list of changed files under a message like _`Oh no! Running the migrate script modified some files:`_.
-> You can also try running the test script locally.
+
+##### Migration Snapshot Failures
```

Comment:
cc @Aslemammad @johnnyreilly - would love to know if this is helpful. Is there anything else you think it should say? Any way to make it more clear?

Reply:
I have been wondering for a while what the migration script actually is / what it does / what its purpose is. Might be nice to have something in there that would cover that?

Reply:
It's good though. Question: is your idea to have a link in errors to this documentation? Or to replicate this in the test failure? trying to think of ways to narrow the gap between "oh something is broken" and "here's how to fix it"

Reply:
> what the migration script actually is / what it does / what its purpose is

That's ... a really good point! I guess it never really was well-documented. Added.

> is your idea to have a link in errors to this documentation? Or ...

I'd like to give users a breadcrumb trail they can follow. I'm averse to duplicating documentation info between docs and test error messages, and don't want the info to be solely containing in test error messages. So I think it follows that the right strategy is to explain the errors in docs and point to the docs from the test error message.